covidapi/import_data.py

```diff
@@ -13,21 +22,17 @@ def main():
     Base.metadata.create_all(engine)
     db_instance = SessionLocal()
 
-    with open(CSV_FILE) as csv_file:
-        csv_reader = csv.reader(csv_file, delimiter=',')
-        next(csv_reader, None)  # skip the headers
-
-        for row in csv_reader:
-            dr = DailyReport(
-                province_state=row[0],
-                country_region=row[1],
-                last_update=datetime.fromisoformat(row[2]),
-                confirmed=row[3],
-                deaths=row[4],
-                recovered=row[5],
-            )
-
-            db_instance.add(dr)
+    for row in get_data_with_caching('02-29-2020'):
```

Comment:
can you please explain how this iterator works? I guess `02-29-2020` is the initial value, but how does it get the "next" date available?

Reply:
At this point it's just fetching a single date, but I changed this in a later commit.

If it's more helpful for reviewing, I could squash some of these commits together?

Reply:
No no, it's fine for now. Just ignore my comments if you think they will be clearly answered later and I will mark those conversation as resolved if I notice the later commit answers my question. Having separate commits makes me easier to read the code.